### PR TITLE
`Development`: Display LTI login name for newly generated Artemis users 

### LINKED ISSUE #7483: `LTI`: Re-Login users when they login to moodle
### Is your feature request related to a problem?

The Artemis user account will be auto-generated, but users can logout via the i-Frame from Artemis.
They might not remember their username which follows the pattern `<coursename>_<firstName>_<lastName>`
Students might forget their username and might not be able to login to Artemis again without contacting the instructors.

### Describe the solution you'd like

1. The user is always logged in to the Artemis Account when logging in to Moodle (this would be the first thing I would try)
2. In the Profile the Artemis login name could be displayed

### Describe alternatives you've considered

_No response_

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by basak-akan**
> [!WARNING]  
> **Deploy only locally or on ts11!** The LTI feature is only available on TS11.

#### General

- [x] This is a small issue that I tested locally and was confirmed by another developer on a test server.
- [x] Language: I followed the [guidelines for inclusive, diversity-sensitive, and appreciative language](https://docs.artemis.cit.tum.de/dev/guidelines/language-guidelines/).
- [x] I chose a title conforming to the [naming conventions for pull requests](https://docs.artemis.cit.tum.de/dev/development-process/#naming-conventions-for-github-pull-requests).
#### Server
- [x] **Important**: I implemented the changes with a very good performance and prevented too many (unnecessary) database calls.
- [x] I followed the [coding and design guidelines](https://docs.artemis.cit.tum.de/dev/guidelines/server/).

#### Client
- [x] **Important**: I implemented the changes with a very good performance, prevented too many (unnecessary) REST calls and made sure the UI is responsive, even with large data.
- [x] I followed the [coding and design guidelines](https://docs.artemis.cit.tum.de/dev/guidelines/client/).
- [x] I added multiple screenshots/screencasts of my UI changes.
- [x] I translated all newly inserted strings into English and German.

### Motivation and Context

Fixes #7483 

### Description

Login name added for generated password pop-up. The predefined login name is filled in while accessing Artemis via an external LMS (e.g. Moodle).

### Steps for Testing

Prerequisites:
-  Please use TS11 to test this PR since lti profile is enabled
- 1 Student

1. Navigate to https://moodle.ase.in.tum.de/
2. Login using one of the provided users below. Please check the used check box so other testers won't test with the same user.
3. Navigate to "LTI Testing Course"
4. Select one of the exercises e.g. "TS11 Authentication Test Exercise"
5. Verify that a pop-up shows up with login and password credentials.
6. Copy the shown username and password to a notepad.
7. Check the acknowledge checkbox and confirm to close the pop-up.
8. Log out from Artemis inside the iframe in Moodle.
9. Reload the page and verify you see the login page.
10. Verify username is filled and the same as the username from the 7th step.

List of sample users for moodle: (users are ready for re-testing again after final fixes)
- [x] username:  jane_doe password: use same password for artemis_test_user_1 
- [x] username:  joey_read password: use same password for artemis_test_user_1
- [x] username:  barney_young password: use same password for artemis_test_user_1 
- [x] username:  john_smith password: use same password for artemis_test_user_1 

#### Code Review
- [x] Code Review 1
- [x] Code Review 2
#### Manual Tests
- [x] Test 1
- [x] Test 2

#### Client

| Class/File | Line Coverage | Confirmation (assert/expect) |
|------------|--------------:|---------------------:|
| [lti13-exercise-launch.component.ts](https://bamboo.ase.in.tum.de/browse/ARTEMIS-TESTS5772/latest/artifact/shared/Coverage-Report-Client-Tests/app/lti/lti13-exercise-launch.component.ts.html) | 100% | ✅ |
| [lti-initializer-modal.component.ts](https://bamboo.ase.in.tum.de/browse/ARTEMIS-TESTS5772/latest/artifact/shared/Coverage-Report-Client-Tests/app/overview/exercise-details/lti-initializer-modal.component.ts.html) | 78.57% | ✅ |
| [lti-initializer.component.ts](https://bamboo.ase.in.tum.de/browse/ARTEMIS-TESTS5772/latest/artifact/shared/Coverage-Report-Client-Tests/app/overview/exercise-details/lti-initializer.component.ts.html) | 100% | ✅ |

#### Server

| Class/File | Line Coverage | Confirmation (assert/expect) |
|------------|--------------:|---------------------:|
| [Lti13LaunchFilter.java](https://bamboo.ase.in.tum.de/browse/ARTEMIS-TESTS5772/latest/artifact/shared/Coverage-Report-Server-Tests/de.tum.in.www1.artemis.security.lti/Lti13LaunchFilter.html) | 86% | ✅ |
| [Lti13Service.java](https://bamboo.ase.in.tum.de/browse/ARTEMIS-TESTS5772/latest/artifact/shared/Coverage-Report-Server-Tests/de.tum.in.www1.artemis.service.connectors.lti/Lti13Service.html) | 93% | ✅ |

### Screenshots

Login name information on pop-up:

image

Prefilled username:

image

### CODE REVIEW

**Review by JanaNF — CHANGES_REQUESTED** [review]
Manual tested with user jane_doe on legacy_ts11. Worked mostly.

Error at step 6: When I tried to copy the Artemis login credentials through the button symbol  always the password copied.

**Review by sarpsahinalp — DISMISSED** [review]
Tested on ts11 with user barney_young. Worked as expected.

File: src/main/java/de/tum/in/www1/artemis/service/connectors/lti/Lti13Service.java [hunk 1]
```diff
@@ -310,4 +310,20 @@ private void createOrUpdateResourceLaunch(Lti13LaunchRequest launchRequest, User
     public void buildLtiResponse(UriComponentsBuilder uriComponentsBuilder, HttpServletResponse response) {
         ltiService.buildLtiResponse(uriComponentsBuilder, response);
     }
+
+    public void buildLtiEmailInUseResponse(HttpServletResponse response, OidcIdToken ltiIdToken) {
+        Optional<String> optionalUsername = artemisAuthenticationProvider.getUsernameForEmail(ltiIdToken.getEmail());
+
+        String sanitizedUsername = getSanitizedUsername(optionalUsername.get());
```
Comment (rriyaldhi) [inline src/main/java/de/tum/in/www1/artemis/service/connectors/lti/Lti13Service.java:317]: shouldn't we check if optionalUsername is present before getting the value?

**Review by rriyaldhi — COMMENTED** [review]
left a question

**Review by milljoniaer — APPROVED** [review]
tested on ts11, with joey_read, works as expected 👍

File: src/main/java/de/tum/in/www1/artemis/service/connectors/lti/Lti13Service.java (hunk 1 above)
Comment (basak-akan) [reply src/main/java/de/tum/in/www1/artemis/service/connectors/lti/Lti13Service.java]: Thank you for the heads up, you are right. I fixed that now.

**Review by milljoniaer — APPROVED** [review]
re-approve 👍

**Review by JanaNF — DISMISSED** [review]
re-approve

**Review by sarpsahinalp — APPROVED** [review]
re-approve

**Review by rriyaldhi — DISMISSED** [review]
Code

**Review by lennart-keller — DISMISSED** [review]
Code lgtm, left one comment. Can be addressed in a different PR.

File: src/main/java/de/tum/in/www1/artemis/service/connectors/lti/Lti13Service.java [hunk 2]
```diff
@@ -310,4 +310,19 @@ private void createOrUpdateResourceLaunch(Lti13LaunchRequest launchRequest, User
     public void buildLtiResponse(UriComponentsBuilder uriComponentsBuilder, HttpServletResponse response) {
         ltiService.buildLtiResponse(uriComponentsBuilder, response);
     }
+
+    public void buildLtiEmailInUseResponse(HttpServletResponse response, OidcIdToken ltiIdToken) {
```
[the remark this answers is not in the document]
Comment (basak-akan) [reply src/main/java/de/tum/in/www1/artemis/service/connectors/lti/Lti13Service.java:314]: Thank you for the feedback. I agree that JavaDoc comments would improve the completeness of the code documentation. I'll make sure to add the JavaDoc comments in a subsequent PR

**Review by milljoniaer — APPROVED** [review]
Reapprove after comment added 👍

**Review by rriyaldhi — APPROVED** [review]

**Review by jakubriegel — APPROVED** [review]
Demonstrated during a testing session - works fine 👍🏻

**Review by laurenzfb — APPROVED** [review]
Tested in testing session. Worked as expected!

**Review by vinceclifford — APPROVED** [review]
Tested during testing session. Works as expected

**Pull request merged**